### PR TITLE
Upgrade template validation to new GH Issue Form

### PR DESCRIPTION
**Pull request by chadwhitacre**
GitHub just rolled out [Issue Forms](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms) to support proper validation! 💃

--------------

https://github.com/getsentry/.github-chadwhitacre/issues/new?template=bug.yml

bug

--------------

https://github.com/getsentry/.github-chadwhitacre/issues/new?template=feature.yml

feature

### CODE REVIEW

**Comment by chadwhitacre** [comment]
P.S. I did consider [auto-applying]() `Status: Untriaged` as part of this, but we only want to do that for external users, so we'll let other ............................... oops. In this repo we want `Status: Unrouted`, which we were doing in `validate-new-issue.yml`. Converting this PR to draft, will pick up next week!

File: .github/workflows/validate-new-issue.yml [hunk 1]
```diff
@@ -1,100 +0,0 @@
...
-              if [ ! -s headings-in-template ]; then
-                echo "No headers in template. 🤷"
-              elif [ "$(comm -12 headings-in-template headings-in-issue)" ]; then
-                echo "Match! 👍 💃"
-                if diff -Bw "$template" issue > /dev/null; then
-                  echo "... like, an /exact/ match. 😖"
-                  REASON="${REASON_EXACT_MATCH}"
-                  break
-                else
-                  gh-issue-label "${issue_number}" "Status: Unrouted"
```
Comment (chadwhitacre) [inline .github/workflows/validate-new-issue.yml:76]: Derp. We still need this.

Comment (chadwhitacre) [reply .github/workflows/validate-new-issue.yml:76]: d0f1bc3d3c10a00134a6114f10e33c9b6ad09eb2

**Comment by chadwhitacre** [comment]
Kicked out d0f1bc3d3c10a00134a6114f10e33c9b6ad09eb2, still calling this a draft until I can test outside this repo.

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+description: Tell us about something that's not working the way we (probably) intend.
+body:
+  - type: dropdown
+    id: environment
+    attributes:
+      label: Environment
+      description: Where are you using our software?
+      options:
+        - SaaS (https://sentry.io/)
+        - single-tenant (if you know, you know)
```
Comment (BYK) [inline .github/ISSUE_TEMPLATE/bug.yml:11]: I'd double check this with support/sales/someone 🙂

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        - self-hosted (`onpremise` deployment)
```
Comment (BYK) [inline .github/ISSUE_TEMPLATE/bug.yml:12]:
```suggestion
        - self-hosted ([`onpremise`](https://github.com/getsentry/onpremise) deployment)
```

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 4]
```diff
@@ -0,0 +1,58 @@
+name: 🐞 Bug Report
+description: Tell us about something that's not working the way we (probably) intend.
+body:
+  - type: dropdown
+    id: environment
+    attributes:
+      label: Environment
+      description: Where are you using our software?
+      options:
```
Comment (BYK) [inline .github/ISSUE_TEMPLATE/bug.yml:9]: Shall we also recognize the "helm chart", and "custom" self-hosted variants.

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        - SaaS (https://sentry.io/)
+        - single-tenant (if you know, you know)
+        - self-hosted (`onpremise` deployment)
+    validations:
+      required: true
+  - type: input
+    id: version
+    attributes:
+      label: Version
+      description: If self-hosted, what version are you running?
```
Comment (BYK) [inline .github/ISSUE_TEMPLATE/bug.yml:19]:
```suggestion
      description: If self-hosted, what version are you running? Check the bottom left corner of any Sentry page you are hosting and you'll see.
```

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    validations:
+      required: false
```
Comment (BYK) [inline .github/ISSUE_TEMPLATE/bug.yml:21]:
```suggestion
    placeholder: 21.7.0
    validations:
      required: false
```

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    validations:
+      required: true
+  - type: textarea
+    id: more-info
+    attributes:
+      label: More Info 
+      description: Logs? Screenshots? Yes, please.
+    validations:
+      required: false
+  - type: markdown
```
Comment (BYK) [inline .github/ISSUE_TEMPLATE/bug.yml:52]: Shall we make use of some behavioral psychology: https://www.pnas.org/content/117/13/7103

```suggestion
  - type: checkboxes
    id: veracity
    attributes:
      label: Confirmation
      options:
        - label: I have filled out this form properly
        - label: I have [searched](https://git.io/JcIay) existing issues on this
  - type: markdown
```

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    attributes:
+      value: |-
+        ## Thanks 🙏
+        Check our [triage docs](https://open.sentry.io/triage/) for what to expect next.
```
Comment (BYK) [inline .github/ISSUE_TEMPLATE/bug.yml:56]: 💯

File: .github/ISSUE_TEMPLATE/feature.yml [hunk 9]
```diff
@@ -0,0 +1,28 @@
+name: 💡 Feature Request
+description: Tell us about a problem our software could solve but doesn't.
```
Comment (BYK) [inline .github/ISSUE_TEMPLATE/feature.yml:2]:
```suggestion
description: Tell us about a problem our software could solve but doesn't, yet.
```

File: .github/ISSUE_TEMPLATE/feature.yml [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+        I want to make whirled peas, but Sentry doesn't blend.
+    validations:
+      required: true
+  - type: textarea
+    id: expected
+    attributes:
+      label: Solution Brainstorm
+      description: We know you have bright ideas to share ... share away, friend.
+      placeholder: |-
+        Add a blender to Sentry.
```
Comment (BYK) [inline .github/ISSUE_TEMPLATE/feature.yml:19]: This should be a separate service like Relay so we can scale.

File: .github/workflows/unroute-new-issue.yml [hunk 11]
```diff
@@ -0,0 +1,33 @@
+name: Unroute new issue
```
Comment (BYK) [inline .github/workflows/unroute-new-issue.yml:1]: We don't even need this (if you are willing to let go of the org-member exception) as we can just do

```yaml
labels: ['Status: Unrouted']
```

inside the new templates and call it a day.

File: .github/ISSUE_TEMPLATE/bug.yml (hunk 2 above)
Comment (chadwhitacre) [reply .github/ISSUE_TEMPLATE/bug.yml:11]: Meh, will remove for now to expedite, it's not on the existing form.

File: .github/ISSUE_TEMPLATE/bug.yml (hunk 3 above)
Comment (chadwhitacre) [reply .github/ISSUE_TEMPLATE/bug.yml:12]: Sadly this is not markdown, just _seems_ like it with the backticks. :-/

File: .github/workflows/unroute-new-issue.yml (hunk 11 above)
Comment (chadwhitacre) [reply .github/workflows/unroute-new-issue.yml:1]: Per https://github.com/getsentry/sentry/pull/26898#issuecomment-868826922, I think the org-member distinction is worth preserving.

File: .github/ISSUE_TEMPLATE/feature.yml (hunk 9 above)
Comment (chadwhitacre) [reply .github/ISSUE_TEMPLATE/feature.yml:2]: Love the optimism but don't want to overcommit our PM/POs. 😁

File: .github/ISSUE_TEMPLATE/bug.yml (hunk 7 above)
Comment (chadwhitacre) [reply .github/ISSUE_TEMPLATE/bug.yml:52]: Overkill. (Also the link is about top vs. bottom of honesty checks ... not, as I would have expected, about whether honesty checks work in the first place.)

Comment (BYK) [reply .github/ISSUE_TEMPLATE/bug.yml:52]: Oh wait, I just read the "significance" section more carefully and seems like this paper is actually refuting both top and bottom veracity checks 🤯

**Comment by chadwhitacre** [comment]
Pushed off d0f1bc3d3c10a00134a6114f10e33c9b6ad09eb2, will handle in

File: .github/ISSUE_TEMPLATE/bug.yml (hunk 2 above)
Comment (chadwhitacre) [reply .github/ISSUE_TEMPLATE/bug.yml]: 2ccdba028a7b9e7741ba0bae5f7a15f2b50e7f93

File: .github/ISSUE_TEMPLATE/bug.yml (hunk 6 above)
Comment (chadwhitacre) [reply .github/ISSUE_TEMPLATE/bug.yml:21]: 2ccdba028a7b9e7741ba0bae5f7a15f2b50e7f93

File: .github/ISSUE_TEMPLATE/bug.yml (hunk 5 above)
Comment (chadwhitacre) [reply .github/ISSUE_TEMPLATE/bug.yml:19]: 2ccdba028a7b9e7741ba0bae5f7a15f2b50e7f93

**Review by BYK — APPROVED** [review]

**Pull request merged**